src/cider/nrepl/middleware/track_state.clj

```diff
@@ -281,21 +345,32 @@
          all-namespaces (if cljs
                           (vals (cljs-ana/all-ns cljs))
                           (all-ns))
-         project-ns-map (fast-reduce (fn [acc ns]
-                                       (let [name (ns-name-fn ns)]
-                                         (if (jar-ns-fn name)
-                                           acc ;; Remove all jar namespaces.
-                                           (assoc! acc name (ns-as-map ns all-namespaces)))))
-                                     all-namespaces)
+         ns-as-map-fn (memoize ns-as-map)
+         project-ns-map (claypoole/with-shutdown! [pool (claypoole/threadpool (if (System/getenv "CI")
```

Comment:
I had to do this, else tests in CI would time out.

I strongly believe it only boils down to Circle servers being slow - just 1 vCPU - adding threads makes it worse.

As a consequence, builds will be slower.

It shouldn't be a practical problem for users, given we'll use 4 threads for them.

There's also great care in avoiding these IO reads for macros that do not need.

Reply:
CI slowness is no more after my second commit.